### PR TITLE
(topN)runtime_predicate is only triggered when the column name is obtained

### LINKED ISSUE #27485: [Bug] Crash when sorting in the outer query using a reference column from the inner subquery
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

release-2.0.2.1 b387ac2

### What's Wrong?

when sort use runtime predicate ,be crashed 
be.out generated the flowing log 

*** Query id: d0695a46cf6a45b8-a177521a64fb2ad3 ***
*** tablet id: 0 ***
*** Aborted at 1700646804 (unix time) try "date -d @1700646804" if you are using GNU date ***
*** Current BE git commitID: b387ac2 ***
*** SIGSEGV address not mapped to object (@0x28) received by PID 139715 (TID 140243 OR 0x7f513f316700) from PID 40; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /data2/doris/be/src/common/signal_handler.h:417
 1# os::Linux::chained_handler(int, siginfo_t*, void*) in /usr/lib/jvm/java-1.8.0/jre/lib/amd64/server/libjvm.so
 2# JVM_handle_linux_signal in /usr/lib/jvm/java-1.8.0/jre/lib/amd64/server/libjvm.so
 3# signalHandler(int, siginfo_t*, void*) in /usr/lib/jvm/java-1.8.0/jre/lib/amd64/server/libjvm.so
 4# 0x00007F525DDAD400 in /lib64/libc.so.6
 5# doris::TabletSchema::column(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const at /data2/doris/be/src/olap/tablet_schema.cpp:938
 6# doris::vectorized::RuntimePredicate::update(doris::vectorized::Field const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool) at /data2/doris/be/src/runtime/runtime_predicate.cpp:162
 7# doris::vectorized::VSortNode::sink(doris::RuntimeState*, doris::vectorized::Block*, bool) in /data2/doris/output/be/lib/doris_be
 8# doris::vectorized::VSortNode::open(doris::RuntimeState*) at /data2/doris/be/src/vec/exec/vsort_node.cpp:190
 9# doris::PlanFragmentExecutor::open_vectorized_internal() in /data2/doris/output/be/lib/doris_be
10# doris::PlanFragmentExecutor::open() at /data2/doris/be/src/runtime/plan_fragment_executor.cpp:271
11# doris::FragmentExecState::execute() at /data2/doris/be/src/runtime/fragment_mgr.cpp:265
12# doris::FragmentMgr::_exec_actual(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::RuntimeState*, doris::Status*)> const&) at /data2/doris/be/src/runtime/fragment_mgr.cpp:534
13# std::_Function_handler<void (), doris::FragmentMgr::exec_plan_fragment(doris::TExecPlanFragmentParams const&, std::function<void (doris::RuntimeState*, doris::Status*)> const&)::$_0>::_M_invoke(std::_Any_data const&) at /var/local/ldb-toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_function.h:291
14# doris::ThreadPool::dispatch_thread() in /data2/doris/output/be/lib/doris_be
15# doris::Thread::supervise_thread(void*) at /data2/doris/be/src/util/thread.cpp:470
16# start_thread in /lib64/libpthread.so.0
17# clone in /lib64/libc.so.6

### What You Expected?

fix it

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by TomaYoko**
…s obtained

## Proposed changes

Issue Number: close #27485

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by TomaYoko** [comment]
run buildall

**Comment by TomaYoko** [comment]
run buildall

**Comment by TomaYoko** [comment]
run buildall

**Comment by LemonLiTree** [comment]
run buildall

**Comment by hello-stephen** [comment]
run clickbench

**Comment by hello-stephen** [comment]
run p0

**Comment by TomaYoko** [comment]
run buildall

File: be/src/vec/exec/vsort_node.cpp [hunk 1]
```diff
@@ -148,11 +148,13 @@ Status VSortNode::sink(RuntimeState* state, vectorized::Block* input_block, bool
             if (!new_top.is_null() && (old_top.is_null() || new_top != old_top)) {
                 auto& sort_description = _sorter->get_sort_description();
                 auto col = input_block->get_by_position(sort_description[0].column_number);
-                bool is_reverse = sort_description[0].direction < 0;
-                auto query_ctx = state->get_query_ctx();
-                RETURN_IF_ERROR(
-                        query_ctx->get_runtime_predicate().update(new_top, col.name, is_reverse));
-                old_top = std::move(new_top);
+                if (!col.name.empty()) {
```
Comment (yiguolei) [inline be/src/vec/exec/vsort_node.cpp:151]: In which scenario, col.name is empty?

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by zhiqiang-hhhh — APPROVED** [review]
LGTM

**Pull request merged**